### PR TITLE
Implement metrics pings

### LINKED ISSUE #47: Implement metrics pings
From the PRD:

> Workflow 3: Statistics
>
> To track the performance and effectiveness of this API, some statistics will be available to administrators:
>
> Lag time from submission to Watchdog (step 1) and submission to PhotoDNA (step 3).
> Processing time from Microsoft (i.e. step 3 to step 4c)
> Statistics about the size of the message queue, both currently and historically.
> Numbers of positives that have not been reviewed and are under review.
> Total numbers of images processed.
>
> Each of these should be available globally, as well as broken down per consumer application.

### PR DESCRIPTION
**Pull request by lmorchard**
- Tweaks to metrics.md to reflect current functionality

- lib/metrics.js to wrap metrics pings

- Add pings to functions

- Mock endpoint that just logs POST requests, used for metrics in dev stack

- Tests

Fixes #47

### CODE REVIEW

**Comment by lmorchard** [comment]
Still have some work to do in adding tests for the metrics pings and seeing how they look when sent to a mock end point. But, wanted to get a work-in-progress PR submitted so folks can see the changes I'm proposing to metrics.md

File: docs/metrics.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Additional fields submitted are described below.
 - *consumer_name*: the name of the consumer submitting the request: string
```
Comment (wresuolc) [inline docs/metrics.md]: Please update the date at the top of this file

File: docs/metrics.md [hunk 2]
```diff
@@ -94,33 +103,16 @@ Example:
...
   "timestamp": "2018-05-18T16:38:33.464Z",
 
-  "event": "worker_works",
+  "event": "workero"
+  "worker_id": "8cdb1e6b-7e15-489d-b171-e7a05781c5da",
   "consumer_name": "screenshots,
   "watchdog_id": "9ad08ec4-be1a-4327-b4ef-282bed37621f"
   "photodna_tracking_id": "1_photodna_a0e3d02b-1a0a-4b38-827f-764acd288c25",
   "is_match": false,
   "is_error": false
```
Comment (wresuolc) [inline docs/metrics.md]: I know it's my mistake, but can you add a comma here? :)

File: docs/metrics.md [hunk 3]
```diff
@@ -94,33 +103,16 @@ Example:
   "topic": "watchdog-proxy",
   "timestamp": "2018-05-18T16:38:33.464Z",
 
-  "event": "worker_works",
+  "event": "workero"
```
Comment (wresuolc) [inline docs/metrics.md]: why did this change?

**Comment by wresuolc** [comment]
r+wc

File: docs/metrics.md (hunk 3 above)
Comment (lmorchard) [reply docs/metrics.md]: Doh! Think that was a typo

**Comment by lmorchard** [comment]
Still working on tests, but going to consider the metrics changes to be okay

**Comment by lmorchard** [comment]
Added tests, squashed, and rebased against master. I think this is in better shape to merge now

File: functions/processQueueItem.js [hunk 4]
```diff
@@ -79,12 +89,25 @@ module.exports.handler = async function({ ReceiptHandle, Body }) {
...
+    await Metrics.workerWorks({
+      consumer_name: user,
+      worker_id: awsRequestId,
+      watchdog_id: id,
+      photodna_tracking_id: upstreamServiceResponse.TrackingId,
+      is_error: false,
+      is_match: upstreamServiceResponse.IsMatch,
+      timing_sent: timingSent,
+      timing_received: timingReceived,
+      timing_submitted: timingSubmitted
```
Comment (chenba) [inline functions/processQueueItem.js]: The timing values don't match what's in the metrics doc.

- It looks like `timing_sent` is meant to be `timing_retrieved`.
- `timing_retrieved` is missing.
- `timing_received` appears to be `timing_sent` and `timing_received` combined. (Although I don't think we can break that request timing down?)

File: lib/metrics.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      consumer_name,
+      watchdog_id,
+      photodna_tracking_id,
+      is_match,
+      is_error,
+      timing_retrieved,
+      timing_sent,
+      timing_received,
+      timing_submitted
+    })
```
Comment (chenba) [inline lib/metrics.js]: The `worker_id` is missing here.

**Pull request merged**